pkg/networkservice/common/heal/server.go

```diff
@@ -193,7 +193,9 @@ func (f *healServer) restoreConnection(ctx context.Context, request *networkserv
 		}
 	}
 
-	f.processHeal(ctx, request.Clone(), opts...)
+	if ctx.Err() == nil {
```

Comment:
As I can see we already have a check on `ctx.Err()` at line 217. Should we remove the check from line 217 with this patch?

Reply:
You mean this for loop condition?
https://github.com/networkservicemesh/sdk/blob/d38c49bc61201044ab9e4060e915171881ba2826/pkg/networkservice/common/heal/server.go#L217

Reply:
If so, no we shouldn't :)

Reply:
Moved this check to `processHeal`.